### PR TITLE
gen_resolvconf.py: handle Docker (EBUSY error) in resolv.conf rename

### PR DESCRIPTION
**Pull request by alberts**

### CODE REVIEW

**Comment by jgehrcke** [comment]
:+1:

**Comment by vespian** [comment]
:+1:

File: packages/spartan/extra/gen_resolvconf.py [hunk 1]
```diff
@@ -87,6 +88,15 @@ def check_server(addr):
...
 # conflicting things like https://dcosjira.atlassian.net/browse/DCOS-305
 
-os.rename(resolvconf_path + ".tmp", resolvconf_path)
+try:
+    os.rename(resolvconf_path + ".tmp", resolvconf_path)
+except OSError as e:
+    # fall back to old behavior because resolv.conf in dcos-docker
+    # is a mount point that doesn't like getting renamed
+    if e.errno == errno.EBUSY:
+        with open(resolvconf_path, 'w') as f:
```
Comment (cmaloney) [inline packages/spartan/extra/gen_resolvconf.py]: Can we add a log message that this is happening?

Comment (alberts) [inline packages/spartan/extra/gen_resolvconf.py]: done

**Comment by cmaloney** [comment]
:ship:

**Pull request merged**